### PR TITLE
Jamila development

### PR DESCRIPTION
**Pull request by jamila-sabaz**
This is all I've done so far, will add a table with projects later tomorrow. Just wanted to try to do the first pull request

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -1,18 +1,104 @@
...
+                    </p>
+                </article>
+            </section>
+
+            <center>
+                <h4>Click here to get a random tv show suggestion from Jamila:</h4>
+                <button onclick="addRandomGreeting()">tv show?</button>
+                <div id="greeting-container"></div>
+            </center>
+
```
Comment (g-wittie) [inline portfolio/src/main/webapp/index.html]: Nitpick: One space to delimit blocks is ok, but avoid having more than one space.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -1,18 +1,104 @@
...
+            <center>
+                <h4>Click here to get a random tv show suggestion from Jamila:</h4>
+                <button onclick="addRandomGreeting()">tv show?</button>
+                <div id="greeting-container"></div>
+            </center>
+            
+            
+            <!-- Table with my projects tbc -->
+
```
Comment (g-wittie) [inline portfolio/src/main/webapp/index.html:57]: Nitpick: No space between the block and the comment, that makes it clearer that the comment describes the block.

**Review by g-wittie — CHANGES_REQUESTED** [review]
Hi Jamila, I've left a couple of nitpick comments (nitpick meaning it's a minor style issue) for you to resolve.

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/index.html]: fixed in the new pushed commit
also added a bit of content but no significant changes

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/index.html:57]: fixed in the new pushed commit
also added a bit of content but no significant changes

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -1,18 +1,104 @@
...
-    <div id="content">
-      <h1>My Portfolio</h1>
-      <p>This is my portfolio.</p>
-      <p>Click here to get a random message:</p>
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
-    </div>
-  </body>
+    <head>
+        <meta charset="UTF-8">
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/index.html]: typically we indent by 2 spaces https://google.github.io/styleguide/htmlcssguide.html#Indentation
(otherwise your indenting is very consistent, congrats.)

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -1,18 +1,104 @@
...
+        <script src="script.js"></script>
+    </head>
+
+    <body>
+        <!-- Image with intro for aesthetics -->
+        <div class="hero-image">
+            <div class="hero-text">
+                <h1>I am Zhamilya (Jamila) Sabazova</h1>
+                <p>And I'm a Software Engineer from Adelaide</p>
+                <button>Learn more about me</button>
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/index.html]: Is this button supposed to do something? 
if yes could you add a TODO(jamila-sabaz): ... comment explaining what it will do, to make it obvious this is an intentional decision.
If no, maybe a button is not the right choice

File: portfolio/src/main/webapp/style.css [hunk 5]
```diff
@@ -1,9 +1,303 @@
...
+
+
+
+
+
+
+
+
+
+/* W3.CSS 4.15 December 2020 by Jan Egil and Borge Refsnes */
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/style.css]: rather than copying and pasting the entire stylesheet in here it's probably better to include their stylesheet by link, or put it in it's own file. keep this file for your own styles.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -1,9 +1,303 @@
...
+    background-color: Tomato;
+    color: rgb(10, 124, 124);
+    height: 100%;
+}
 #content {
-  margin-left: auto;
-  margin-right: auto;
-  width: 650px;
+  width: 300px;
+
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/style.css]: as max mentioned in the html file try and avoid unnecessary whitespace, one line between {} sections is sufficient.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -17,7 +17,7 @@
  */
 function addRandomGreeting() {
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: This is not really adding a greeting anymore, I think this function needs a rename

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -26,3 +26,18 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+// Automatic Slideshow - change image every 3 seconds
+var myIndex = 0;
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: We generally try to avoid using global variables where possible, it's really easy to make complicated and hard to debug interactions with them, so it's good style to avoid them.
Maybe try instead giving carousel an argument and bind that argument in setTimeout.

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -26,3 +26,18 @@ function addRandomGreeting() {
...
+carousel();
+
+function carousel() {
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: Try to add a verb into your function names to make it clear what it's doing to the carousel
https://google.github.io/styleguide/jsguide.html#naming-method-names

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -26,3 +26,18 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+// Automatic Slideshow - change image every 3 seconds
+var myIndex = 0;
+carousel();
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: It's not great practice to call functions like this, as you have no control over when they get called, it's just whenever the js file gets loaded, which might be before the dom is ready.
A better way to do this is to wait for the event that says the dom is loaded and then call this function.
eg:
document.addEventListener('DOMContentLoaded', (event) => {
    displayCarouselAtIndex(0);
});
https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -26,3 +26,18 @@ function addRandomGreeting() {
...
+
+function carousel() {
+  var i;
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: use let instead of var, var has similar problems to global variables https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

**Review by lorikeet-s — CHANGES_REQUESTED** [review]
looking good, added a few style comments.

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/index.html:4]: fixed

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: fixed - changed name of the function and all its variable names in all files

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/index.html]: now actually properly fixed

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/index.html]: fixed the button, now it takes the user to about me section
added the same button that takes the user to projects section

File: portfolio/src/main/webapp/style.css (hunk 5 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/style.css]: fixed - removed that whole chunk from my doc, used a link to online library using href="w3school.css" to use it as an additional style sheet

File: portfolio/src/main/webapp/style.css (hunk 6 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/style.css]: fixed - removed redundant spaces

File: portfolio/src/main/webapp/script.js (hunk 10 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: fixed - used it in a new carousel but it's not automatic

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: fixed - passing an integer to the function

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: fixed - new name is showSlides

File: portfolio/src/main/webapp/script.js (hunk 11 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: fixed - using let instead of var inside the function

Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: I actually fixed these details earlier today, but got to reply on comments just now :)

**Comment by jamila-sabaz** [comment]
Annoying challenge I can't fix: In About me section I put the containers with info and photos into buttons but if I click all of them sequentially the blocks appear in a column but I want them to be in a row. :((( not sure if it's css issue and if it is possible to put them in a row (horizontally).
Next challenge: interactive navigation bar and additional pages - contacts, etc

File: portfolio/src/main/webapp/html/contact.html [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+                <i class="fa fa-bars"></i>
+            </a>
+        </div>
+			
+        <!-- Main Content! -->
+        <div id="content">    
+            <h3>Contact Form</h3>
+
+                <div class="w3-content w3-container w3-center">
+                <form action="/action_page.php">
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/html/contact.html]: I don't think you have any php in this project, you probably want this to be a function in your javascript.
it looks like you've copied and pasted this from somewhere, please don't just copy things, looking at examples is fine, but then write it yourself to make sure you understand it.

File: portfolio/src/main/webapp/html/contact.html [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+		<title>Jamila's Portfolio</title>
+		<link rel="stylesheet" href="style.css">
+		<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+        <script src="script.js"></script>
+	</head>
+
+	<body>
+        <div class="topnav" id="myTopnav">
+            <a href="#home" class="active">Home</a>
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/html/contact.html]: this navbar doesn't work, mostly because contact.html is in a subfolder and also because you're trying to link to ids that aren't in this page, they're in teh index page, you'd need to do something like index.html#home I think

File: portfolio/src/main/webapp/index.html [hunk 14]
```diff
@@ -1,104 +1,196 @@
...
-                    <img src="./images/google2018.jpeg" alt="mypathtogoogle" style="width:100%">
-                    <p>In February 2019 my mum and I went to travel to USA. We did a road trip from Los Angeles to San Francisco and then to Las Vegas, but on the way to SF we visited San Jose to check out some science museums and Silicon Valley campuses. 
-                        Then we realised a tour can be arranged only by a company representative, who we did not know, so we just stared at it from the outside (lol). This is me on Google HQ Mountain View campus, I came there as a tourist and I told my mum I wanted to work there, she was skeptical. 
-                        It’s been almost 2 years since then and now I’m a STEP intern at Google Sydney branch. Dreams come true! 
-                    </p>
-                </article>
+	<body>
+    <div class="topnav" id="myTopnav">
+        <a href="#home" class="active">Home</a>
+        <a href="#reflections">My Reflections Blog</a>
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/index.html]: This doesn't seem to go anywhere?

File: portfolio/src/main/webapp/html/contact.html [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+            
+        </div>
+​
+        <!-- Footer -->
+        <footer class="w3-container w3-padding-64 w3-center w3-black w3-xlarge">
+            <a href="https://github.com/jamila-sabaz" class="w3-button w3-bar-item">GitHub</a>
+            <a href="https://www.linkedin.com/in/jamila-sabazova/" class="w3-button w3-bar-item">Linkedin</a>
+        </footer>
+	</body>
+</html>
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/html/contact.html]: add a blank new line at the end of files (the other html, css, and js files in this PR too)

File: portfolio/src/main/webapp/script.js [hunk 16]
```diff
@@ -15,29 +15,95 @@
...
+    if (factHasBeenClicked === true){
+      factContainer.innerText = null;
+      factHasBeenClicked = false;
+    }
+  else{
+      factContainer.innerText = fact;
+      factHasBeenClicked = true;
+  }
+}
+//Button
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: Can you give this a more descriptive function comment please.

File: portfolio/src/main/webapp/script.js [hunk 17]
```diff
@@ -15,29 +15,95 @@
...
+    section.className = section.className.replace(" w3-show", "");
+  } 
+}
+/**
+ * Slideshow 
+ */
+document.addEventListener('DOMContentLoaded', () => {
+showSlides(0);
+});
+function plusSlides(n) {
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js:95]: Can you give these functions function comments.

File: portfolio/src/main/webapp/script.js [hunk 18]
```diff
@@ -15,29 +15,95 @@
...
-// Automatic Slideshow - change image every 3 seconds
-var myIndex = 0;
-carousel();
 
-function carousel() {
-  var i;
-  var x = document.getElementsByClassName("mySlides");
-  for (i = 0; i < x.length; i++) {
-     x[i].style.display = "none";
+function navBar() {
```
Comment (lorikeet-s) [inline portfolio/src/main/webapp/script.js]: and this one.

**Review by lorikeet-s — CHANGES_REQUESTED** [review]
You added a few commits that add a bunch of new features, specifically the navbar and the contact page.
a) in future please don't add features to a PR that's already in review, just create a new PR.
b) these features don't seem to work, I think it would be better if you rolled them back and left a working page given we're supposed to be going onto a new week's project. Max is the git guru he can talk you through doing that.

File: portfolio/src/main/webapp/html/contact.html (hunk 12 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/html/contact.html]: removed this html page

File: portfolio/src/main/webapp/html/contact.html (hunk 13 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/html/contact.html]: noted! but the file is removed

File: portfolio/src/main/webapp/index.html (hunk 14 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/index.html]: removed nav bar

File: portfolio/src/main/webapp/html/contact.html (hunk 15 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/html/contact.html]: fixed

File: portfolio/src/main/webapp/script.js (hunk 16 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: added to the new commit

File: portfolio/src/main/webapp/script.js (hunk 18 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js]: removed this function when reversed changes

File: portfolio/src/main/webapp/script.js (hunk 17 above)
Comment (jamila-sabaz) [reply portfolio/src/main/webapp/script.js:95]: added more comments
